tools/shared.py

```diff
@@ -2336,14 +2336,10 @@ def check_call(cmd, *args, **kw):
     raise
 
 def suffix(name):
-  parts = name.split('.')
-  if len(parts) > 1:
-    return parts[-1]
-  else:
-    return None
+  return os.path.splitext(name)[1]
```

Comment:
Oh.. good call.  I will update this PR.

However, while investigating I noticed that we should probably followup with another PR because it looks like suffix() is only being used emcc.py and only in two places:
emcc.py:1550  -> here it looks like its mis-used because its expecting the '.'
emcc.py:338 -> here it looks like the code would be simpler if we include the '.'

Perhaps a followup?